packages/core/src/rag/index.ts

```diff
@@ -742,13 +742,34 @@ Thank you and happy classifying!`;
 
     const similaritySearchTerm = topic ? `${title}. ${topic}` : title;
 
-    const result = await vectorStore.similaritySearchWithScore(
-      similaritySearchTerm,
-      k * 5, // search for more records than we need
-    );
+    let result:
```

Comment:
This removes the error about join([]) that langchain mistakenly throws when it tries to join on an empty dataset